crs_motion_planning/src/freespace_planning_server.cpp

```diff
@@ -0,0 +1,192 @@
+
+    std::string urdf_path, srdf_path;
+    urdf_path = ament_index_cpp::get_package_share_directory("crs_support") + "/urdf/crs.urdf";
+    srdf_path = ament_index_cpp::get_package_share_directory("crs_support") + "/urdf/ur10e_robot.srdf";
+    std::stringstream urdf_xml_string, srdf_xml_string;
+    std::ifstream urdf_in(urdf_path);
+    urdf_xml_string << urdf_in.rdbuf();
+    std::ifstream srdf_in(srdf_path);
+    srdf_xml_string << srdf_in.rdbuf();
+
+    tesseract_local_ = std::make_shared<tesseract::Tesseract>();
+    tesseract_scene_graph::ResourceLocator::Ptr locator = std::make_shared<tesseract_rosutils::ROSResourceLocator>();
+    tesseract_local_->init(urdf_xml_string.str(), srdf_xml_string.str(), locator);
+  }
+private:
+  void jointCallback(const sensor_msgs::msg::JointState::SharedPtr joint_msg)
+  {
+    curr_joint_state_ = *joint_msg;
+  }
+  void planService(std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Request> request,
+                    std::shared_ptr<crs_msgs::srv::CallFreespaceMotion::Response> response)
+  {
+    std::string target_frame = request->target_link;
+    std::string base_link_frame = "world";
+
+    Eigen::Isometry3d tcp_eigen;
+    tcp_eigen.setIdentity();
+
+    auto traj_pc = std::make_shared<tesseract_motion_planners::TrajOptPlannerFreespaceConfig>(tesseract_local_, "manipulator", target_frame, tcp_eigen);
+
+    // Initialize vector of target waypoints
+    std::vector<tesseract_motion_planners::Waypoint::Ptr> trgt_wypts;
+
+    // Define initial waypoint
+    Eigen::VectorXd init_joint_pose(6);
+    init_joint_pose << curr_joint_state_.position[0], curr_joint_state_.position[1], curr_joint_state_.position[2], curr_joint_state_.position[3], curr_joint_state_.position[4], curr_joint_state_.position[5];
+    tesseract_motion_planners::JointWaypoint::Ptr joint_init_waypoint = std::make_shared<tesseract_motion_planners::JointWaypoint>(init_joint_pose, joint_name_list_);
+    joint_init_waypoint->setIsCritical(false);
+
+    // Define goal waypoint
+    Eigen::Vector3d goal_pose(request->goal_pose.translation.x, request->goal_pose.translation.y, request->goal_pose.translation.z);
+    Eigen::Quaterniond goal_ori(request->goal_pose.rotation.w, request->goal_pose.rotation.x, request->goal_pose.rotation.y, request->goal_pose.rotation.z);
+    tesseract_motion_planners::CartesianWaypoint::Ptr goal_waypoint = std::make_shared<tesseract_motion_planners::CartesianWaypoint>(goal_pose, goal_ori);
+
+    goal_waypoint->setIsCritical(true);
+    Eigen::VectorXd coeffs(6);
+    coeffs << 10, 10, 10, 10, 10, 10;
+    goal_waypoint->setCoefficients(coeffs);
+
+    // Add waypoints
+    trgt_wypts.push_back(joint_init_waypoint);
+    trgt_wypts.push_back(goal_waypoint);
+
+    // Write target waypoints to trajopt planning config
+    traj_pc->target_waypoints = trgt_wypts;
+
+    // Various freespace configuration settings
+    traj_pc->num_steps = 200;
```

Comment:
I think we may want to expose these values as configurable ros parameters in order to reconfigure without needing to recompile